FE/angular/src/app/components/movies/movies.component.ts

```diff
@@ -1,36 +1,38 @@
 import { Component, OnInit } from '@angular/core';
-import { Observable, tap } from 'rxjs';
+import { BehaviorSubject, Observable, map, mergeMap, switchMap } from 'rxjs';
 import { DataService, MovieComplete, MovieData } from '../../services/data.service';
 
+interface MovieState {
+  selectedDecade: number;
+  decades: number[];
+  filteredMovies: MovieComplete[];
+  allMovies: MovieComplete[];
+}
+
 @Component({
   selector: 'app-movies',
   templateUrl: './movies.component.html'
 })
 export class MoviesComponent implements OnInit {
-  public currDecade: number | undefined;
-  // public decades: number[] = [];
-  public filteredMovies: MovieComplete[] = [];
-  public movies: MovieComplete[] = [];
-  public moviesSubscription: Observable<MovieData> = {} as Observable<MovieData>;
+  public state$: Observable<MovieState> = {} as Observable<MovieState>;
+  public selectedDecade$: BehaviorSubject<number> = new BehaviorSubject<number>(0);
 
   constructor(private dataService: DataService) {}
 
   public ngOnInit(): void {
-    this.moviesSubscription = this.dataService.getMovies().pipe(
-      tap((data) => {
-        this.movies = data.Search;
-        this.displayMovies();
+    this.state$ = this.selectedDecade$.pipe(
+      switchMap((selectedDecade: number) => {
```

Comment:
Using the `switchMap` is good BUT it means that 
* you will reload the movie EACH time the decade changes (not good)
* if you ever wanted the use to have a movie search option (to reload movies based on search param) this would fail.

Reply:
You should have used `combineLatest()`

Reply:
Sure, I will change this.

Reply:
I've learnt the power of RxJs, now i'll definitely make use of this power in my upcoming learning and development journey.
Thanks a lot.